GoogleForCreators/web-stories-wp
# Implemented new filter for categories and tags to story block

## Context

Two more filters are to be added to the story block (latest stories).

## Summary

It would be nice to have support to the custom taxonomies “web_story_category” and "web_story_tag" for story block.
The displayed stories should be limited to the selected categories/tags.

See also:
https://wordpress.org/support/topic/feature-new-block-type-to-support-categories/

## Relevant Technical Choices

The component /packages/stories-block/src/block/components/authorSelection.js was used as a model and two more components (categorySelection.js & tagSelection.js) were added. The block attributes and the PHP method `get_query_args` were adjusted accordingly .

## To-do

The components authorSelection.js, categorySelection.js and tagSelection.js are very similary.
Maybe it would be a good idea to unify them in the future.

Translations for two new phrases:
- __('Tags', 'web-stories')
- __('Categories', 'web-stories')

## User-facing changes

In the story block (latest stories), two additional filters for categories and keywords have been added below the filter for authors.

## Testing Instructions


This PR can be tested by following these steps:

1. Compile sources and modify plugin.
2. Select story block with variation "latest stories".
3. Select a category/tag similar to the author filter.

The block should display the filtered stories in Gutenberg and after saving on frontend.

## Reviews

### Does this PR have a security-related impact?

No.

### Does this PR change what data or activity we track or use?

No.

### Does this PR have a legal-related impact?

No.

## Checklist
